dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -413,3 +416,47 @@ dashboard.header.updateTimestamp = function () {
     $('.project_updated_at').text("Not saved"); // TODO i18n
   }
 };
+
+function getSummarizedProgressForAnonymousUser () {
+  var summarizedProgress = {};
+  var levelProgress = {};
+  var cookie = document.cookie.split(';');
+  
+  for(var i = 0; i < cookie.length; i++) {
+    var pair = cookie[i].split('=');
+    var key = pair[0].trim();
+
+    switch(key) {
+      case 'progress':
+        var allLevelsProgress = JSON.parse(decodeURIComponent(pair[1]));
+        for(var level in allLevelsProgress) {
+          if (allLevelsProgress.hasOwnProperty(level)) {
+            var individualLevelProgressDescription = '';
+            var individualLevelProgress = allLevelsProgress[level];
+
+            //TODO: Find a way to use the constants defined somewhere
+            if (individualLevelProgress >= 1000) {
+              individualLevelProgressDescription = 'submitted';
+            } else if (individualLevelProgress >= 30) {
+              individualLevelProgressDescription = 'perfect';
+            } else if (individualLevelProgress >= 20) {
+              individualLevelProgressDescription = 'passed';
+            } else {
+              individualLevelProgressDescription = 'attempted';
```

Comment:
Although it probably shouldn't happen, you should do something sensible for individualLevelProgress = 0, which is not even attempted.